### ISSUE TITLE
how alpha get collaborated in backpropagation phase

### ISSUE THREAD

**Issue by saria85** [hugochan/KATE#17]
I hope it becomes my last question :octocat: 

So, I am clear on the theoretical concept, but while in implementation I can not get how keras can get the alpha to participate in the back propagation.
Actually I know my question is not related mainly to your implementation but the way that keras works.

From what I know when keras back propagates, it tries to update weight matrices automatically, I could not find any resource how can it gets other neurons which already being set to zero to participate with alpha percentage.

I hope I could give some hint what I mean.

Thanks for taking the time:)

**Comment by saria85** [comment]
I am curious on this question :)

**Comment by hugochan** [comment]
@saria85 It is a good question! I don't think it is a question only related to Keras, any library which does back propagation should be able to work as expected.

I think you can consider it in this way. You can image that you have a residual connection, y = alpha * x + f(x), even though you make f(x) = 0, you can still back propagate through x because of the alpha * x term. Similarly, in KATE, we reset loser neurons, but the energy of loser neurons get passed to the output via the alpha amplification component.

Hope this helps.

**Comment by hugochan** [comment]
@saria85 This might help https://drive.google.com/file/d/17FCuRz2emU7CJPgAASTInBi9zH4EZtNS/view?usp=sharing

**Comment by saria85** [comment]
Thank you I will go through that then.

**Comment by saria85** [comment]
Actually, Apart from your source, I went through http://neuralnetworksanddeeplearning.com/chap2.html
 but it was the same thing I already have known. I still have difficulty understanding 1. how the neuron can be zero in the feedforward step and then they will contribute to updating weight in the backward step?
2. particularly, if they(zero neurons) can contribute how alpha will be assigned to them not other neurons. my point is exactly this sentence of the paper:

Note that in the back-propagation procedure, the
gradients will first flow through the winner neurons in the hidden
layer and then the loser neurons via the α amplification connections.

I did not find anything in the implementation to understand how alpha just first go to winner neuron and then loser neuron via alpha amplification. from the way generally backpropagation works, neuron with zero weight will not participate in the updating weight, so I thought maybe you have changed the implementation of the network.
Please let me know if I am missing something :)

Thanks
